64_MergeSort/main.cpp

```diff
@@ -0,0 +1,368 @@
+#include "nbl/application_templates/MonoDeviceApplication.hpp"
+#include "nbl/application_templates/MonoAssetManagerAndBuiltinResourceApplication.hpp"
+
+using namespace nbl;
+using namespace core;
+using namespace system;
+using namespace asset;
+using namespace video;
+
+// A high level overview of the algorithm implementation (merge sort).
+// This will be a X pass algorithm, where X is log2(number_of_elements).
+// In each phase, you have to 'merge' 2 sorted list of elements into a single list.
+// Each of these 'lists' will have thier number of elements roughly doubled in each phase (after the merge step of previous phase).
```

Comment:
I was kinda expecting you to roll up the CPU logic into a separate class + shared HLSL/c++ header to drive it (see the counting sort code)